src/shared/components/Animations/InfiniteScroll.stories.tsx

```diff
@@ -0,0 +1,362 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import { withInfo } from '@storybook/addon-info';
+import {
+  withKnobs,
+  text,
+  array,
+  object,
+  boolean,
+  number,
+} from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+import InfiniteScroll from './InfiniteScroll';
+import ListItem from './ListItem';
+import { FetchableState } from '../../store/reducers/utils';
+import { PaginatedList } from '@bbp/nexus-sdk';
+
+// @ts-ignore
+const shuffleArray = (arr: any[]) =>
+  arr
+    .map((a: any) => [Math.random(), a])
+    .sort((a: any, b: any) => a[0] - b[0])
+    .map((a: any) => a[1]);
+
+const exampleItems = [
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/InVitroSliceReconstructedPatchedNeuron',
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/ReconstructedCell',
+  'http://www.w3.org/ns/prov#Entity',
+  'https://bbp-nexus.epfl.ch/vocabs/bbp/neurosciencegraph/core/v0.1.0/ReconstructedPatchedCell',
+  'http://www.w3.org/ns/dcat#Dataset',
+];
+
+const mockServer = (numberOfPages: number, pageSize: number) => {
+  let position = 0;
+  let items = exampleItems;
+  return {
+    next: () => {
+      if (position >= numberOfPages) {
+        return;
+      }
+      position = position + 1;
+      items = shuffleArray(exampleItems);
+    },
+    getItems: () => items,
+  };
+};
+
+storiesOf('Components/InfiniteScroll/Basic', module)
+  .addDecorator(withKnobs)
+  .add(
+    'InfiniteScroll',
+    withInfo(`
+    Scroll all the way down and ask for more!
+
+    ~~~js
+    <InfiniteScroll
+            makeKey={item => item}
+            itemComponent={(item: any, index: number) => <p>{item}</p>}
+            items={items}
+            total={items.length}
+            next={action('pagination-change')}
+            loading={loading}
+          />
+  />
+    ~~~
+  `)(() => {
+      return React.createElement(() => {
+        const numberOfItems = number('number of items', 200);
+        const isLoading = boolean('force loading?', false);
+        const fetchablePaginatedList = object('fetachblePaginatedList', {
+          isFetching: isLoading,
+          error: null,
+          data: {
+            results: exampleItems,
+            total: numberOfItems,
+            index: 0,
+          },
+        });
+        const next = () => {
+          action('next')();
+        };
+        return (
+          <div style={{ margin: '50px 40px 0px' }}>
+            <h2>InfiniteScroll</h2>
+            <InfiniteScroll
+              makeKey={item => item}
+              itemComponent={(item: any, index: number) => {
+                return <ListItem label={item} id={`${item}-${index}`} />;
+              }}
+              next={next}
+              fetchablePaginatedList={fetchablePaginatedList}
+            />
+          </div>
+        );
+      });
+    })
+  );
+
+storiesOf('Components/InfiniteScroll/Simulated/OnClick', module)
```

Comment:
This is quite a story!! :open_mouth:

Reply:
Yeah, but I want to make the story page clear except for one example each... not sure another good way to do that